File: app/src/main/java/com/datagene/daggerexample22/ui/fragment/ProfileFragment.java
package com.datagene.daggerexample22.ui.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProviders;


import com.datagene.daggerexample22.R;
import com.datagene.daggerexample22.ui.AnotherActivity;
import com.datagene.daggerexample22.viewmodels.ViewModelProviderFactory;

import javax.inject.Inject;

import dagger.android.support.DaggerFragment;

public class ProfileFragment extends DaggerFragment {

    @Inject
    ViewModelProviderFactory providerFactory;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view= inflater.inflate(R.layout.fragment_main, container, false);
        view.findViewById(R.id.button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getActivity(), AnotherActivity.class);
                startActivity(intent);
            }
        });
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        ProfileFragmentViewModel viewModel = ViewModelProviders.of(this, providerFactory).get(ProfileFragmentViewModel.class);
        Log.d("zzzz", "fragment viewmodel is created");
    }
}

File: app/src/main/java/com/datagene/daggerexample22/ui/fragment/ProfileFragment.kt
package com.datagene.daggerexample22.ui.fragment

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.lifecycle.ViewModelProviders
import com.datagene.daggerexample22.R
import com.datagene.daggerexample22.ui.AnotherActivity
import com.datagene.daggerexample22.viewmodels.ViewModelProviderFactory
import dagger.android.support.DaggerFragment
import javax.inject.Inject

class ProfileFragment : DaggerFragment() {
    @JvmField
    @Inject
    var providerFactory: ViewModelProviderFactory? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_main, container, false)
        view.findViewById<View>(R.id.button).setOnClickListener {
            val intent = Intent(activity, AnotherActivity::class.java)
            startActivity(intent)
        }
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        val viewModel = ViewModelProviders.of(this, providerFactory).get(ProfileFragmentViewModel::class.java)
        Log.d("zzzz", "fragment viewmodel is created")
    }
}